src/Modules/OrchardCore.Commerce/Controllers/ShoppingCartController.cs

```diff
@@ -109,13 +110,9 @@ public async Task<ActionResult> AddItem(ShoppingCartLineUpdateModel line, string
                 await _shoppingCartSerializer.ParseCartLineAsync(line),
                 storeIfOk: true);
 
-            if (_workflowManager != null)
-            {
-                await _workflowManager.TriggerEventAsync(
-                    nameof(ProductAddedToCartEvent),
-                    new { LineItem = parsedLine },
-                    "ShoppingCart-" + _shoppingCartPersistence.GetUniqueCartId(shoppingCartId));
-            }
+            await _workflowManagers.TriggerEventAsync<ProductAddedToCartEvent>(
```

Comment:
OrchardCore.Workflows is not required and we still use `IEnumerable<IWorkflowManager> _workflowManagers` instead of `IWorkflowManager _workflowManager`. The empty checking has been abstracted into the extension method [here](https://github.com/Lombiq/Helpful-Libraries/blob/4a0828dc015ec98f0501165b8f9d754b7522a783/Lombiq.HelpfulLibraries.OrchardCore/Workflow/WorkflowManagerExtensions.cs#L59) (which uses the new [`InvokeFirstOrCompletedAsync`](https://github.com/Lombiq/Helpful-Libraries/blob/4a0828dc015ec98f0501165b8f9d754b7522a783/Lombiq.HelpfulLibraries.Common/Extensions/EnumerableExtensions.cs#L392) extension method) to avoid code repetition.

Reply:
I see, thanks!